binr/r2r/r2r.c

```diff
@@ -631,3 +646,277 @@ static void print_log(R2RState *state, ut64 prev_completed, ut64 prev_paths_comp
+		R2RTestResultInfo *result = *it;
+		if (result->test->type != R2R_TEST_TYPE_CMD) {
+			continue;
+		}
+		if (result->test->path != edited_file) { // this works because all the paths come from the string pool
+			continue;
+		}
+		R2RCmdTest *test = result->test->cmd_test;
+		test->run_line += delta;
+
+#define DO_KEY_STR(key, field) \
+		if (test->field.value) { \
+			if (test->field.line_begin >= start_line) { \
+				test->field.line_begin += delta; \
+			} \
+			if (test->field.line_end >= start_line) { \
+				test->field.line_end += delta; \
+			} \
+		}
+
+#define DO_KEY_BOOL(key, field) \
+		if (test->field.set && test->field.line >= start_line) { \
+			test->field.line += delta; \
+		}
+
+		R2R_CMD_TEST_FOREACH_RECORD(DO_KEY_STR, DO_KEY_BOOL)
+#undef DO_KEY_STR
+#undef DO_KEY_BOOL
+	}
+}
+
+static void replace_cmd_kv(const char *path, ut64 line_begin, ut64 line_end, const char *key, const char *value, RPVector *fixup_results) {
+	char *content = r_file_slurp (path, NULL);
+	if (!content) {
+		eprintf ("Failed to read file \"%s\"\n", path);
+		return;
+	}
+	char *kv = format_cmd_kv (key, value);
+	if (!kv) {
+		free (content);
+		return;
+	}
+	ut64 kv_lines = r_str_char_count (kv, '\n');
+	char *newc = replace_lines (content, line_begin, line_end, kv);
+	free (kv);
+	free (content);
+	if (!newc) {
+		return;
+	}
+	if (r_file_dump (path, (const ut8 *)newc, -1, false)) {
+#if __UNIX__
+		sync ();
+#endif
+		ut64 lines_before = line_end - line_begin;
+		st64 delta = (st64)kv_lines - lines_before;
+		if (line_end == line_begin) {
+			delta++;
+		}
+		fixup_tests (fixup_results, path, line_end, delta);
+	} else {
+		eprintf ("Failed to write file \"%s\"\n", path);
+	}
+	free (newc);
+}
+
+static void interact_fix(R2RTestResultInfo *result, RPVector *fixup_results) {
+	assert (result->test->type == R2R_TEST_TYPE_CMD);
+	R2RCmdTest *test = result->test->cmd_test;
+	R2RProcessOutput *out = result->proc_out;
+	if (test->expect.value && out->out) {
+		replace_cmd_kv (result->test->path, test->expect.line_begin, test->expect.line_end, "EXPECT", out->out, fixup_results);
+	}
+	if (test->expect_err.value && out->err) {
+		replace_cmd_kv (result->test->path, test->expect_err.line_begin, test->expect_err.line_end, "EXPECT_ERR", out->err, fixup_results);
+	}
+}
+
+static void interact_break(R2RTestResultInfo *result, RPVector *fixup_results) {
+	assert (result->test->type == R2R_TEST_TYPE_CMD);
+	R2RCmdTest *test = result->test->cmd_test;
+	ut64 line_begin;
+	ut64 line_end;
+	if (test->broken.set) {
+		line_begin = test->broken.set;
+		line_end = line_begin + 1;
+	} else {
+		line_begin = line_end = test->run_line;
+	}
+	replace_cmd_kv (result->test->path, line_begin, line_end, "BROKEN", "1", fixup_results);
+}
+
+static void interact_commands(R2RTestResultInfo *result, RPVector *fixup_results) {
+	assert (result->test->type == R2R_TEST_TYPE_CMD);
+	R2RCmdTest *test = result->test->cmd_test;
+	if (!test->cmds.value) {
+		return;
+	}
+	char *name;
```

Comment:
maybe worth = NULL; in theory when returning -1 name will not be initialized, so the free below it will crash